core/Tracker/GoalManager.php

```diff
@@ -195,6 +207,49 @@ public function detectGoalMatch($goal, Action $action)
         return $action->getActionUrl();
     }
 
+    private function detectNumericGoalMatch($goal, Action $action, VisitProperties $visitProperties)
+    {
+        switch ($goal['match_attribute']) {
+            case 'visit_duration':
+                $valueToMatchAgainst = ((int) $visitProperties->getProperty('visit_total_time')) / 60;
+                break;
+            case 'visit_total_actions':
+                $valueToMatchAgainst = $visitProperties->getProperty('visit_total_actions') ?: 0;
+                $valueToMatchAgainst = $valueToMatchAgainst + 1; // to account for current action
+                break;
+            case 'visit_total_pageview':
+                $valueToMatchAgainst = $this->getPageviewCountInVisit($visitProperties->getProperty('idvisit'));
```

Comment:
Any chance to delay this to Matomo 4 where we might have a `visit_total_pageviews`? I would not want to have a query here.

Reply:
I guess we can just ignore goals like this in the GA importer? I suppose it will be easy enough to add support later, some users will just have to wait.

In the meantime, what about visit duration goals? Is that ok for 3.11?

Reply:
visit duration should be fine? Not sure how accurate `visit_total_time` is though. Might be more accurate to do `$request->getCurrentTime() - visit_first_action_time` ?

Reply:
Sounds safer.

Reply:
Updated.